dask_kubernetes/operator/controller/controller.py

```diff
@@ -389,6 +376,92 @@ async def daskcluster_create_components(
 
     patch.status["phase"] = "Pending"
 
+@kopf.on.update("daskcluster.kubernetes.dask.org")
+async def daskcluster_update(
+    spec: kopf.Spec,
+    status: kopf.Status, 
+    meta: kopf.Meta,
+    name: str | None,    
+    namespace: str | None,
+    diff: kopf.Diff,     
+    patch: kopf.Patch,   
+    logger: kopf.Logger,
+    **__: Any
+):
+    """When the DaskCluster resource is updated update all the components."""
+    assert name
+    assert namespace
```

Comment:
If either of these assertions fail then kopf will retry automatically forever. We probably need to catch the assertion error and just return if this isn't what we want to hapen.